### PR TITLE
Migrate openshift installer upi

### PR DESCRIPTION
**Pull request by alvaroaleman**
Based on top of https://github.com/openshift/ci-tools/pull/1330

/hold

To check on rehearsals in https://github.com/openshift/release/pull/13217

### CODE REVIEW

File: cmd/determinize-ci-operator/main.go [hunk 1]
```diff
@@ -180,10 +189,72 @@ func migrateOpenshiftInstallerCustomTestImageTemplates(
...
+
+	for idx, test := range configuration.Configuration.Tests {
+		if test.OpenshiftInstallerUPIClusterTestConfiguration == nil ||
+			(len(allowedCloudproviders) != 0 && !allowedCloudproviders.Has(string(test.OpenshiftInstallerUPIClusterTestConfiguration.ClusterProfile))) {
+			continue
+		}
+
+		commandFields := strings.Fields(test.Commands)
+		if n := len(commandFields); n != 2 {
+			return 0, fmt.Errorf("command %q didn't have exactly two fields", test.Commands)
```
Comment (petr-muller) [inline cmd/determinize-ci-operator/main.go:221]: I don't think this needs to be error-level. If we have strange tests like this, I think it's fine to just skip them for now, maybe warn about them.

File: cmd/determinize-ci-operator/main.go [hunk 2]
```diff
@@ -180,10 +189,72 @@ func migrateOpenshiftInstallerCustomTestImageTemplates(
...
+		}
+		equalSignSplit := strings.Split(commandFields[0], "=")
+		if n := len(equalSignSplit); n != 2 {
+			return 0, fmt.Errorf("splitting first field of command %q by = didn't yield exactly two results", test.Commands)
```
Comment (petr-muller) [inline cmd/determinize-ci-operator/main.go:225]: Same here.

File: cmd/determinize-ci-operator/main.go [hunk 3]
```diff
@@ -180,10 +189,72 @@ func migrateOpenshiftInstallerCustomTestImageTemplates(
...
+		}
+
+		var testCommandEnv string
+		switch commandFields[1] {
+		case "run-tests":
+			testCommandEnv = ""
+		case "run-upgrade":
+			testCommandEnv = "run-upgrade"
+		default:
+			return 0, fmt.Errorf("command %q has unrecognized command element %q, known elements: ['run-tests', 'run-upgrade']", test.Commands, commandFields[1])
```
Comment (petr-muller) [inline cmd/determinize-ci-operator/main.go:235]: And here too :)

File: cmd/determinize-ci-operator/main.go (hunk 1 above)
Comment (alvaroaleman) [reply cmd/determinize-ci-operator/main.go]: Updated

**Review by petr-muller — APPROVED** [review]

**Comment by alvaroaleman** [comment]
@petr-muller seems prow missed your approval
/hold cancel

**Pull request merged**